feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/era/StakingEraInteractorFactory.kt

```diff
@@ -0,0 +1,25 @@
+package io.novafoundation.nova.feature_staking_impl.domain.era
+
+import io.novafoundation.nova.feature_staking_impl.data.parachainStaking.RoundDurationEstimator
+import io.novafoundation.nova.feature_staking_impl.data.parachainStaking.repository.CurrentRoundRepository
+import io.novafoundation.nova.feature_staking_impl.domain.StakingInteractor
+import io.novafoundation.nova.runtime.ext.isParachain
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+
+class StakingEraInteractorFactory(
+    private val roundDurationEstimator: RoundDurationEstimator,
+    private val currentRoundRepository: CurrentRoundRepository,
+    private val stakingInteractor: StakingInteractor
+) {
+
+    fun create(chain: Chain): StakingEraInteractor {
+        return if (chain.isParachain) {
+            ParachainStakingEraInteractor(
+                roundDurationEstimator,
+                currentRoundRepository
+            )
+        } else {
+            RelaychainStakingEraInteractor(stakingInteractor)
+        }
+    }
```

Comment:
Why is it not a part of `StartStakingInteractor` and correpsondign factory? 
Also, it is not true in general that if chain is parachain it will have `parachainStaking`. Many chains use their own staking implementations